src/library/interface_mono.cpp

```diff
@@ -9,20 +9,46 @@ OrbSlam2InterfaceMono::OrbSlam2InterfaceMono(const ros::NodeHandle& nh,
     : OrbSlam2Interface(nh, nh_private) {
   // Getting data and params
   subscribeToTopics();
-  //advertiseTopics();
-  //getParametersFromRos();
+
+  if(use_body_transform_){
+    if(!getBodyTransform()){
+      ros::shutdown();
+      exit(1);
+    }
+  }
+
   // Creating the SlAM system
   slam_system_ = std::shared_ptr<ORB_SLAM2::System>(
       new ORB_SLAM2::System(vocabulary_file_path_, settings_file_path_,
-                            ORB_SLAM2::System::MONOCULAR, true));
+                            ORB_SLAM2::System::MONOCULAR, visualization_));
 }
 
 void OrbSlam2InterfaceMono::subscribeToTopics() {
   // Subscribing to the required data topics
-  image_sub_ = nh_.subscribe("camera/image_raw", 1,
+  image_sub_ = nh_.subscribe("cam0/image_raw", 1,
                              &OrbSlam2InterfaceMono::imageCallback, this);
 }
 
+bool OrbSlam2InterfaceMono::getBodyTransform()
+{
+
+  cv::FileStorage fsSettings(settings_file_path_, cv::FileStorage::READ);
+  if(!fsSettings.isOpened())
+  {
+      ROS_ERROR("ERROR: Wrong path to settings");
+      return false;
+  }
+
+  cv::Mat T_C0_B;
+  Transformation T_C_B;
+
+  fsSettings["T_c0_fcuimu"] >> T_C0_B;
+  convertOrbSlamPoseToKindr(T_C0_B, &T_C_B);
+  T_B_C_ = T_C_B.inverse();
+
+  return true;
+}
```

Comment:
There are a few ways to load parameters. Our policy is to try to use the ROS parameter server exclusively. A good example of how to do this can be found here:

https://github.com/ethz-asl/voxblox/blob/master/voxblox_ros/launch/euroc_dataset_bag.launch#L62